program_intent_engagement/apps/api/serializers.py

```diff
@@ -1,4 +1,27 @@
-# Serializers that can be shared across multiple versions of the API
-# should be created here. As the API evolves, serializers may become more
-# specific to a particular version of the API. In this case, the serializers
-# in question should be moved to versioned sub-package.
+
+from program_intent_engagement.apps.core.models import ProgramIntent
+from rest_framework import serializers
+from program_intent_engagement.apps.core.models import ProgramIntent, CERTAINTY
+
+class ProgramIntentSerializer(serializers.ModelSerializer):
+    """
+    Serializer for the Program Intent Model
+    """
+
+    # TODO: gotta figure out the JWT stuff for this
+    user = serializers.CharField(required=True)
```

Comment:
I'm not sure that I understand what [source](https://www.django-rest-framework.org/api-guide/fields/#source) means, is it just a filler?

When you mention we need the LMS user ID, not the local one, what do you mean? I can setup a [SlugRelatedField](https://www.django-rest-framework.org/api-guide/relations/#slugrelatedfield) that would take the LMS_User_Id from the User.

Reply:
that looks like it would do what we want but what a terrible name for a field
I think you could also do source like serializers.IntField(source="user.lms_user_id") if there is an IntField, I'm just guessing. Source seems to just be a path. Or a function, you could write serializers.IntField(source=get_out_the_useful_user_id) and go write that function but it might be overkill.

It looks like the advantage of the SlugRelated is that you could run it in both directions so that serializer could take in a program intent and find the appropriate user by the lms_user_id and set up the DB correctly. But we don't need that here.